latest-issue/src/main/java/net/androidweekly/latestissue/LatestIssueViewModel.kt

```diff
@@ -87,11 +86,14 @@ class LatestIssueViewModel(private val issuesRepository: IssuesRepository) : Bas
         return items?.subList(0, nextTitleIndex)?.toTypedArray()
     }
 
-    fun setItems(items: Array<out Parcelable>?) {
-        this.items = (items as? Array<Item>?)?.toList()
-        viewModelScope.launch(Dispatchers.Main) {
-            _issuesResourceLiveData.value = Resource.Success(items).ignoreElement()
-            _issuesResourceLiveData.value = Resource.Loading(show = false)
+    fun setItems(item: Issue) {
+        items = item.getItems().toList()
+        if (items != null) {
+            showProgress()
+            viewModelScope.launch(Dispatchers.Main) {
+                _issuesResourceLiveData.value = Resource.Success(items).ignoreElement()
+                hideProgress()
+            }
```

Comment:
The old implementation should be reverted, and you should put list of items instead of put one issue